netbox/dcim/filtersets.py

```diff
@@ -1511,22 +1511,16 @@ class Meta:
         ]
 
     def filter_device(self, queryset, name, value):
-        try:
-            devices = Device.objects.filter(**{'{}__in'.format(name): value})
-            vc_interface_ids = []
-            for device in devices:
-                vc_interface_ids.extend(device.vc_interfaces().values_list('id', flat=True))
-            return queryset.filter(pk__in=vc_interface_ids)
-        except Device.DoesNotExist:
-            return queryset.none()
-
-    def filter_device_id(self, queryset, name, id_list):
         # Include interfaces belonging to peer virtual chassis members
         vc_interface_ids = []
         try:
-            devices = Device.objects.filter(pk__in=id_list)
+            devices = Device.objects.filter(**{'{}__in'.format(name): value})
```

Comment:
btw f-strings are faster than str. format() because f-strings are evaluated at compile-time rather than at runtime.

Reply:
I was just following the established convention.  I would say if we switch to f-strings we should do it on all filtersets at once.